lib/cldr/export/data/numbers.rb

```diff
@@ -7,108 +7,181 @@ class Numbers < Base
-              end
+            if (aliased = select_single(format_node, "alias"))
+              format_length_key = format_length_node["type"]&.to_sym || default_format_length_type
+              result[format_key] = xpath_to_default_currency_format_length_node_alias(aliased["path"], number_system, format_length_key)
             else
-              aliased = select_single(format_length_node, "alias")
+              pattern_nodes = select_single(format_node, "pattern")
+              result[format_key] = pattern_nodes.content
+            end
+          end
+          result
+        end
+
+        def parse_currency_format_length_node(format_length_node, type)
+          result = {}
+          select(format_length_node, "#{type}Format").each do |format_node|
+            format_key = format_node["type"]&.to_sym || default_format_type
+
+            result[format_key] ||= select(format_node, "pattern").each_with_object({}) do |pattern_node, pattern_result|
+              pattern_key = pattern_node["type"]&.to_sym || default_pattern_type
+              pattern_count = pattern_node["count"]&.to_sym
+
+              if pattern_count
+                if pattern_result[pattern_key].nil?
+                  pattern_result[pattern_key] ||= {}
+                elsif !pattern_result[pattern_key].is_a?(Hash)
+                  raise "can't parse patterns with and without 'count' attribute in the same section"
+                end
 
-              if aliased
-                format_result[format_key] = xpath_to_redirect(aliased.attribute("path").value)
+                pattern_result[pattern_key][pattern_count] = pattern_node.content
+              else
+                pattern_result[pattern_key] = pattern_node.content
               end
             end
           end
-
-          result[:default] = result[:default][:default] if result[:default]
           result
         end
 
-        def xpath_to_redirect(xpath)
-          length = xpath[/(\w+)FormatLength/, 1]
-          type   = xpath[/@type='(\w+)'/, 1]
-
-          :"numbers.formats.#{length}.patterns.#{type}"
+        def default_format_length_type
+          # TODO: It would be better is this were one of the valid values for the type attribute
+          # <!ATTLIST decimalFormatLength type (full | long | medium | short) #IMPLIED >
+          # But I haven't been able to figure out what the default is.
+          @default_format_length_type ||= :default
         end
 
-        def number_system(type)
-          node = select("numbers/#{type}Formats").first
-          begin
-            node.attribute("numberSystem").value
-          rescue
-            "latn"
+        def default_format_type
+          @default_format_type ||= begin
+            # Verify that the default format type has not changed / is the same for all the types
+            ldml_dtd_file = File.read("vendor/cldr/common/dtd/ldml.dtd")
+            ["currency", "decimal", "percent", "scientific"].each do |type|
```

Comment:
You could potentially make this list a frozen constant, and you could also use this same constant to loop through when building the children in `number_systems`.